sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -435,6 +435,11 @@ object SQLConf {
     defaultValue = Some(true),
     doc = "When false, we will treat bucketed table as normal table")
 
+  val ORDER_BY_ORDINAL = booleanConf("spark.sql.orderByOrdinal",
+    defaultValue = Some(true),
```

Comment:
Should we turn it off by default? It's not a standard feature and hive also turn it off by default.